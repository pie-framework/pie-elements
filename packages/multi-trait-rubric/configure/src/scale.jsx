import React from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';
import ArrowBackIosIcon from '@material-ui/icons/ArrowBackIos';
import ArrowForwardIosIcon from '@material-ui/icons/ArrowForwardIos';

import { withDragContext } from '@pie-lib/drag';
import { color } from '@pie-lib/render-ui';

import TraitsHeader from './traitsHeader';
import TraitTile from './trait';
import {
  Arrow,
  BlockWidth,
  DragHandleSpace,
  MultiTraitButton,
  PrimaryBlockWidth
} from './common';
import {
  DecreaseMaxPoints,
  DeleteScale,
  DeleteTrait,
} from './modals';

const AdjustedBlockWidth = BlockWidth + (2 * 8); // 8 is padding

const styles = {
  scaleWrapper: {
    display: 'flex',
    flexDirection: 'column',
    margin: '12px 0',
    wordBreak: 'break-word',
    padding: `16px 0 16px ${DragHandleSpace}px`,
    position: 'relative'
  },
  maxPoints: {
    width: '300px',
    margin: '16px 0 32px'
  },
  trait: {
    background: color.secondaryBackground(),
    margin: '16px 0',
    padding: '16px'
  }
};

export class Scale extends React.Component {
  state = {
    showDecreaseMaxPointsDialog: false,
    showDeleteScaleDialog: false,
    showDeleteTraitDialog: false,
    currentPosition: 0,
    showRight: null,
    showLeft: null
  };

  set = (newState) => this.setState(newState);

  componentDidMount() {
    if (this.state.showRight === null && this.secondaryBlockRef) {
      this.setState({ showRight: this.secondaryBlockRef.scrollWidth - this.secondaryBlockRef.offsetWidth });
    }
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    if (nextProps.excludeZero !== this.props.excludeZero ||
      nextProps.showStandards !== this.props.showStandards ||
      nextProps.showDescription !== this.props.showDescription) {
      this.setState({
        currentPosition: 0,
        showLeft: false,
        showRight: this.secondaryBlockRef && this.secondaryBlockRef.scrollWidth - this.secondaryBlockRef.offsetWidth
      });
    }
  }

  // Max Points
  updateMaxPointsFieldValue = ({ target }) => {
    const { scale, scaleIndex, onScaleChanged } = this.props || {};
    const { maxPoints, scorePointsLabels, traits } = scale;

    const numberValue = parseInt(target.value, 10);

    this.setState({
      currentPosition: 0,
      showLeft: false,
      showRight: this.secondaryBlockRef && this.secondaryBlockRef.scrollWidth - this.secondaryBlockRef.offsetWidth
    });

    if (numberValue < maxPoints) {
      this.showDecreaseMaxPointsModal({ newMaxPoints: numberValue });
    } else {
      for (let i = 0; i <= numberValue; i++) {
        if (!scorePointsLabels[i]) {
          scorePointsLabels.push('');
        }

        traits.forEach((trait, index) => {
          if (!traits[index].scorePointsDescriptors[i]) {
            traits[index].scorePointsDescriptors.push('');
          }
        });

      }

      onScaleChanged(scaleIndex, { maxPoints: numberValue, scorePointsLabels, traits })
    }
  }

  showDecreaseMaxPointsModal = ({ newMaxPoints }) => this.set({ showDecreaseMaxPointsDialog: true, newMaxPoints });

  hideDecreaseMaxPointsModal = () => this.set({ showDecreaseMaxPointsDialog: false, newMaxPoints: undefined });

  changeMaxPoints = () => {
    const { newMaxPoints } = this.state || {};
    const { scaleIndex, onScaleChanged } = this.props || {};

    if (newMaxPoints) {
      onScaleChanged(scaleIndex, { maxPoints: newMaxPoints });
    }

    this.hideDecreaseMaxPointsModal();
  }

  // Delete Scale
  showDeleteScaleModal = () => this.set({ showDeleteScaleDialog: true });

  hideDeleteScaleModal = () => this.set({ showDeleteScaleDialog: false });

  deleteScale = () => {
    const { scaleIndex, onScaleRemoved } = this.props || {};

    this.hideDeleteScaleModal();

    onScaleRemoved(scaleIndex);
  }

  // Delete Trait
  showDeleteTraitModal = (traitToDeleteIndex) => this.set({ showDeleteTraitDialog: true, traitToDeleteIndex });

  hideDeleteTraitModal = () => this.set({ showDeleteTraitDialog: false, traitToDeleteIndex: undefined });

  onTraitRemoved = () => {
    const { traitToDeleteIndex } = this.state;
    const { scale, scaleIndex, onScaleChanged } = this.props || {};
    let { traits } = scale || {};

    if (traitToDeleteIndex < 0 || traitToDeleteIndex >= traits.length) return;

    traits = [
      ...traits.slice(0, traitToDeleteIndex),
      ...traits.slice(traitToDeleteIndex + 1)
    ];

    onScaleChanged(scaleIndex, { traits });

    this.hideDeleteTraitModal();
  }

  onTraitAdded = () => {
    const { scale, scaleIndex, onScaleChanged } = this.props || {};
    const { traits, scorePointsLabels } = scale || {};

    traits.push({
      name: '',
      description: '',
      standards: [],
      scorePointsDescriptors: Array.from(
        { length: scorePointsLabels.length },
        () => ''
      ),
    });

    onScaleChanged(scaleIndex, { traits });
  }

  onTraitChanged = (traitIndex, trait) => {
    const { scale, scaleIndex, onScaleChanged } = this.props || {};
    const { traits } = scale || {};

    if (traitIndex >= 0 && traitIndex < traits.length) {
      traits[traitIndex] = trait;

      onScaleChanged(scaleIndex, { traits });
    }
  }

  onTraitDropped = (source, newIndex) => {
    const { scale, scaleIndex, onScaleChanged } = this.props || {};
    const { traits } = scale || {};
    const { index: oldIndex } = source;
    const cup = traits[oldIndex];

    const remainingTraits = traits.filter((item, index) => index !== oldIndex);

    const newTraits = [
      ...remainingTraits.slice(0, newIndex),
      cup,
      ...remainingTraits.slice(newIndex),
    ];

    onScaleChanged(scaleIndex, { traits: newTraits });
  }

  decreasePosition = () => {
    const { currentPosition } = this.state;
    const decreasedPosition = currentPosition - (currentPosition === AdjustedBlockWidth / 2 ? AdjustedBlockWidth / 2 : AdjustedBlockWidth);

    this.setState({
      currentPosition: decreasedPosition,
      showRight: decreasedPosition < this.secondaryBlockRef.scrollWidth - this.secondaryBlockRef.offsetWidth,
      showLeft: decreasedPosition > 0
    });
  }

  increasePosition = () => {
    const { currentPosition } = this.state;
    const increasedPosition = currentPosition + (currentPosition === 0 ? AdjustedBlockWidth / 2 : AdjustedBlockWidth);

    this.setState({
      currentPosition: increasedPosition,
      showRight: increasedPosition < this.secondaryBlockRef.scrollWidth - this.secondaryBlockRef.offsetWidth,
      showLeft: increasedPosition > 0
    });
  }

  render() {
    const {
      classes,
      scale,
      scaleIndex,
      showStandards,
      onScaleChanged,
      excludeZero,
      showDescription,
      showLevelTagInput,
      showScorePointLabels,
      enableDragAndDrop
    } = this.props || {};
    const {
      maxPoints,
      scorePointsLabels,
      traitLabel,
      traits
    } = scale || {};

    const {
      showDecreaseMaxPointsDialog,
      showDeleteScaleDialog,
      showDeleteTraitDialog,
      currentPosition,
      showRight,
      showLeft
    } = this.state;

    const scorePointsValues = [];

    // determining the score points values
    for (let pointValue = maxPoints; pointValue >= excludeZero ? 1 : 0; pointValue -= 1) {
      scorePointsValues.push(pointValue);
    }


    return (
      <div
        key={`scale-${scaleIndex}`}
        className={classes.scaleWrapper}
        ref={ref => {
          this.scaleWrapper = ref;
        }}
      >
        <TraitsHeader
          setSecondaryBlockRef={ref => {
            if (ref) {
              this.secondaryBlockRef = ref;
            }
          }}
          key={'header-key'}
          traitLabel={traitLabel}
          scorePointsValues={scorePointsValues}
          scorePointsLabels={scorePointsLabels}
          onScaleChange={(params) => onScaleChanged(scaleIndex, params)}
          onTraitLabelChange={label => onScaleChanged(scaleIndex, { traitLabel: label })}
          showStandards={showStandards}
          showDescription={showDescription}
          showLevelTagInput={showLevelTagInput}
          showScorePointLabels={showScorePointLabels}
          maxPoints={maxPoints}
          updateMaxPointsFieldValue={this.updateMaxPointsFieldValue}
          scaleIndex={scaleIndex}
          showDeleteScaleModal={this.showDeleteScaleModal}
          currentPosition={currentPosition}
        />

        {traits.map((trait, index) => (
          <TraitTile
            key={index}
            index={index}
            trait={trait}
            scorePointsValues={scorePointsValues}
            scorePointsLabels={scorePointsLabels}
            onTraitRemoved={() => this.showDeleteTraitModal(index)}
            onTraitChanged={trait => this.onTraitChanged(index, trait)}
            onTraitDropped={this.onTraitDropped}
            showStandards={showStandards}
            showDescription={showDescription}
            currentPosition={currentPosition}
            enableDragAndDrop={enableDragAndDrop}
          />
        ))}

        <Arrow
          width={`${AdjustedBlockWidth / 2}px`}
          show={showLeft}
          onClick={this.decreasePosition}
          left={`${PrimaryBlockWidth + DragHandleSpace}px`}
          showLevelTagInput={showLevelTagInput}
        >
          <ArrowBackIosIcon/>
        </Arrow>
        <Arrow
          width={`${AdjustedBlockWidth / 2}px`}
          show={showRight}
          onClick={this.increasePosition}
          showLevelTagInput={showLevelTagInput}
        >
          <ArrowForwardIosIcon/>
        </Arrow>

        <MultiTraitButton onClick={this.onTraitAdded}>
          <div dangerouslySetInnerHTML={{ __html: `Add ${traitLabel || ' Trait'}` }}/>
        </MultiTraitButton>

        <DecreaseMaxPoints
          open={!!showDecreaseMaxPointsDialog}
          deleteScorePoints={this.changeMaxPoints}
          cancel={this.hideDecreaseMaxPointsModal}
        />

        <DeleteScale
          open={!!showDeleteScaleDialog}
          scaleIndex={scaleIndex}
          deleteScale={this.deleteScale}
          cancel={this.hideDeleteScaleModal}
        />

        <DeleteTrait
          open={!!showDeleteTraitDialog}
          deleteTrait={this.onTraitRemoved}
          cancel={this.hideDeleteTraitModal}
        />
      </div>
    );
  }
}

Scale.propTypes = {
  classes: PropTypes.object,
  scale: PropTypes.shape({
    maxPoints: PropTypes.number,
    scorePointsLabels: PropTypes.arrayOf(PropTypes.string),
    traitLabel: PropTypes.string,
    traits: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string,
      standards: PropTypes.arrayOf(PropTypes.string),
      scorePointsDescriptors: PropTypes.arrayOf(PropTypes.string),
      description: PropTypes.string,
    }))
  }),
  excludeZero: PropTypes.bool,
  scaleIndex: PropTypes.number,
  onScaleChanged: PropTypes.func,
  onScaleRemoved: PropTypes.func,
  showStandards: PropTypes.bool,
  showLevelTagInput: PropTypes.bool,
  showDescription: PropTypes.bool,
  showScorePointLabels: PropTypes.bool,
  enableDragAndDrop: PropTypes.bool,
}

export default withDragContext(withStyles(styles)(Scale));
