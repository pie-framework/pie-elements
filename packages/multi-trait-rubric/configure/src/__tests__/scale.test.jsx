import { shallow } from 'enzyme';
import React from 'react';
import { Scale } from '../scale';

const scale = () => ({
  excludeZero: false,
  maxPoints: 2,
  scorePointsLabels: ['Non-Scorable', 'Unsatisfactory', 'Satisfactory'],
  traitLabel: 'Category',
  traits: [
    {
      name: 'Presentation',
      standards: [],
      description: '',
      scorePointsDescriptors: [
        'Handwriting is unreadable, or response is blank, not in English, or too brief to evaluate. ',
        'Handwriting poor\n' + '\n' + 'Overall appearance is distracting to unacceptable',
        'Handwriting is generally legible\n' + '\n' + 'Overall appearance is acceptable or better',
      ],
    },
  ]
});

jest.mock('@pie-lib/render-ui', () => ({
  color: {
    secondaryBackground: jest.fn().mockReturnValue('grey'),
    text: jest.fn().mockReturnValue('black'),
  }
}));

describe('Scale', () => {
  let w;

  const wrapper = extras => {
    const defaults = {
      classes: {},
      scale: scale(),
      scaleIndex: 0,
      showStandards: true,
      ...extras
    };
    return shallow(<Scale {...defaults} />);
  };

  describe('snapshot', () => {
    it('renders', () => {
      w = wrapper();

      expect(w).toMatchSnapshot();
    });

    it('renders without traits', () => {
      w = wrapper({ traits: [] });

      expect(w).toMatchSnapshot();
    });

    it('renders without standards', () => {
      w = wrapper({ showStandards: false });

      expect(w).toMatchSnapshot();
    });

    it('renders without descriptions', () => {
      w = wrapper({ showDescription: false });

      expect(w).toMatchSnapshot();
    });

    it('renders with drag and drop enabled', () => {
      w = wrapper({ dragAndDrop: true });

      expect(w).toMatchSnapshot();
    });
  });

  describe('logic', () => {
    let onScaleChanged;

    beforeEach(() => {
      onScaleChanged = jest.fn();
      w = wrapper({ onScaleChanged });
    });


    describe('updateMaxPointsFieldValue', () => {
      it('shows alert box is number less then max points', () => {
        const { maxPoints } = w.instance().props.scale;

        w.instance().updateMaxPointsFieldValue({ target: { value: maxPoints - 1 } });

        expect(w.instance().state.newMaxPoints).toEqual(maxPoints - 1);
      });

      it('changes max points if number more then max points', () => {
        const { maxPoints } = w.instance().props.scale;

        w.instance().updateMaxPointsFieldValue({ target: { value: maxPoints + 1 } });

        expect(onScaleChanged).toBeCalledWith(0, { maxPoints: maxPoints + 1 });
      });
    });

    describe('changeMaxPoints', () => {
      it('removes zero', () => {
        w.instance().setState({ newMaxPoints: 10 });
        w.instance().changeMaxPoints();

        expect(onScaleChanged).toBeCalledWith(0, { maxPoints: 10 });
      });
    });

    describe('deleteScale', () => {
      it('calls onRemoveScale', () => {
        const onScaleRemoved = jest.fn();

        w = wrapper({ onScaleRemoved });
        w.instance().deleteScale();

        expect(onScaleRemoved).toBeCalledWith(0);
      });
    });

    describe('onTraitRemoved', () => {
      it('does not call onScaleChanged if index less than zero', () => {
        w.instance().setState({ traitToDeleteIndex: -1 });
        w.instance().onTraitRemoved();

        expect(onScaleChanged).not.toBeCalled();
      });

      it('does not call onScaleChanged if index more than length', () => {
        w.instance().setState({ traitToDeleteIndex: 100 });
        w.instance().onTraitRemoved();

        expect(onScaleChanged).not.toBeCalled();
      });

      it('calls onScaleChanged', () => {
        w.instance().state.traitToDeleteIndex = 0;
        w.instance().onTraitRemoved();

        expect(onScaleChanged).toBeCalledWith(0, { traits: [] });
      });
    });

    describe('onTraitChanged', () => {
      it('does not call onScaleChanged if index less than zero', () => {
        w.instance().onTraitChanged(-100, {});

        expect(onScaleChanged).not.toBeCalled();
      });

      it('does not call onScaleChanged if index more than length', () => {
        w.instance().onTraitChanged(1000, {});

        expect(onScaleChanged).not.toBeCalled();
      });

      it('calls onScaleChanged', () => {
        w.instance().onTraitChanged(0, {});

        expect(onScaleChanged).toBeCalledWith(0, { traits: [{}] });
      });
    });

    describe('onTraitDropped', () => {
      it('calls onScaleChanged', () => {
        w.instance().onTraitAdded();
        w.instance().onTraitAdded();

        const { traits } = w.instance().props.scale;
        const length = traits.length;
        const lastButOne = traits[length - 2];
        const last = traits[length - 1];

        w.instance().onTraitDropped({ index: length - 2 }, length - 1);

        expect(onScaleChanged).toBeCalledWith(0, {
          traits: [
            ...traits.slice(0, length - 2),
            last,
            lastButOne
          ]
        });
      });
    });
  });
});
