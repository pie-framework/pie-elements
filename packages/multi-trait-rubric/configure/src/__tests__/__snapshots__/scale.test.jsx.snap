// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scale snapshot renders 1`] = `
<div
  key="scale-0"
>
  <WithStyles(TraitsHeaderTile)
    currentPosition={0}
    key="header-key"
    maxPoints={2}
    onScaleChange={[Function]}
    onTraitLabelChange={[Function]}
    scaleIndex={0}
    scorePointsLabels={
      Array [
        "Non-Scorable",
        "Unsatisfactory",
        "Satisfactory",
      ]
    }
    scorePointsValues={Array []}
    setSecondaryBlockRef={[Function]}
    showDeleteScaleModal={[Function]}
    showStandards={true}
    traitLabel="Category"
    updateMaxPointsFieldValue={[Function]}
  />
  <DropTarget(DragSource(WithStyles(TraitTile)))
    currentPosition={0}
    index={0}
    key="0"
    onTraitChanged={[Function]}
    onTraitDropped={[Function]}
    onTraitRemoved={[Function]}
    scorePointsLabels={
      Array [
        "Non-Scorable",
        "Unsatisfactory",
        "Satisfactory",
      ]
    }
    scorePointsValues={Array []}
    secondaryBlockWidth={320}
    showStandards={true}
    trait={
      Object {
        "description": "",
        "name": "Presentation",
        "scorePointsDescriptors": Array [
          "Handwriting is unreadable, or response is blank, not in English, or too brief to evaluate. ",
          "Handwriting poor

Overall appearance is distracting to unacceptable",
          "Handwriting is generally legible

Overall appearance is acceptable or better",
        ],
        "standards": Array [],
      }
    }
  />
  <WithStyles(Component)
    left="232px"
    onClick={[Function]}
    show={null}
    width="108px"
  >
    <pure(ArrowBackIosIcon) />
  </WithStyles(Component)>
  <WithStyles(Component)
    onClick={[Function]}
    show={null}
    width="108px"
  >
    <pure(ArrowForwardIosIcon) />
  </WithStyles(Component)>
  <WithStyles(Component)
    onClick={[Function]}
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "Add Category",
        }
      }
    />
  </WithStyles(Component)>
  <WithStyles(RawDecreaseMaxPoints)
    cancel={[Function]}
    deleteScorePoints={[Function]}
    open={false}
  />
  <WithStyles(RawDeleteScale)
    cancel={[Function]}
    deleteScale={[Function]}
    open={false}
    scaleIndex={0}
  />
  <WithStyles(RawDeleteTrait)
    cancel={[Function]}
    deleteTrait={[Function]}
    open={false}
  />
</div>
`;

exports[`Scale snapshot renders with drag and drop enabled 1`] = `
<div
  key="scale-0"
>
  <WithStyles(TraitsHeaderTile)
    currentPosition={0}
    key="header-key"
    maxPoints={2}
    onScaleChange={[Function]}
    onTraitLabelChange={[Function]}
    scaleIndex={0}
    scorePointsLabels={
      Array [
        "Non-Scorable",
        "Unsatisfactory",
        "Satisfactory",
      ]
    }
    scorePointsValues={Array []}
    setSecondaryBlockRef={[Function]}
    showDeleteScaleModal={[Function]}
    showStandards={true}
    traitLabel="Category"
    updateMaxPointsFieldValue={[Function]}
  />
  <DropTarget(DragSource(WithStyles(TraitTile)))
    currentPosition={0}
    index={0}
    key="0"
    onTraitChanged={[Function]}
    onTraitDropped={[Function]}
    onTraitRemoved={[Function]}
    scorePointsLabels={
      Array [
        "Non-Scorable",
        "Unsatisfactory",
        "Satisfactory",
      ]
    }
    scorePointsValues={Array []}
    secondaryBlockWidth={320}
    showStandards={true}
    trait={
      Object {
        "description": "",
        "name": "Presentation",
        "scorePointsDescriptors": Array [
          "Handwriting is unreadable, or response is blank, not in English, or too brief to evaluate. ",
          "Handwriting poor

Overall appearance is distracting to unacceptable",
          "Handwriting is generally legible

Overall appearance is acceptable or better",
        ],
        "standards": Array [],
      }
    }
  />
  <WithStyles(Component)
    left="232px"
    onClick={[Function]}
    show={null}
    width="108px"
  >
    <pure(ArrowBackIosIcon) />
  </WithStyles(Component)>
  <WithStyles(Component)
    onClick={[Function]}
    show={null}
    width="108px"
  >
    <pure(ArrowForwardIosIcon) />
  </WithStyles(Component)>
  <WithStyles(Component)
    onClick={[Function]}
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "Add Category",
        }
      }
    />
  </WithStyles(Component)>
  <WithStyles(RawDecreaseMaxPoints)
    cancel={[Function]}
    deleteScorePoints={[Function]}
    open={false}
  />
  <WithStyles(RawDeleteScale)
    cancel={[Function]}
    deleteScale={[Function]}
    open={false}
    scaleIndex={0}
  />
  <WithStyles(RawDeleteTrait)
    cancel={[Function]}
    deleteTrait={[Function]}
    open={false}
  />
</div>
`;

exports[`Scale snapshot renders without descriptions 1`] = `
<div
  key="scale-0"
>
  <WithStyles(TraitsHeaderTile)
    currentPosition={0}
    key="header-key"
    maxPoints={2}
    onScaleChange={[Function]}
    onTraitLabelChange={[Function]}
    scaleIndex={0}
    scorePointsLabels={
      Array [
        "Non-Scorable",
        "Unsatisfactory",
        "Satisfactory",
      ]
    }
    scorePointsValues={Array []}
    setSecondaryBlockRef={[Function]}
    showDeleteScaleModal={[Function]}
    showDescription={false}
    showStandards={true}
    traitLabel="Category"
    updateMaxPointsFieldValue={[Function]}
  />
  <DropTarget(DragSource(WithStyles(TraitTile)))
    currentPosition={0}
    index={0}
    key="0"
    onTraitChanged={[Function]}
    onTraitDropped={[Function]}
    onTraitRemoved={[Function]}
    scorePointsLabels={
      Array [
        "Non-Scorable",
        "Unsatisfactory",
        "Satisfactory",
      ]
    }
    scorePointsValues={Array []}
    secondaryBlockWidth={320}
    showDescription={false}
    showStandards={true}
    trait={
      Object {
        "description": "",
        "name": "Presentation",
        "scorePointsDescriptors": Array [
          "Handwriting is unreadable, or response is blank, not in English, or too brief to evaluate. ",
          "Handwriting poor

Overall appearance is distracting to unacceptable",
          "Handwriting is generally legible

Overall appearance is acceptable or better",
        ],
        "standards": Array [],
      }
    }
  />
  <WithStyles(Component)
    left="232px"
    onClick={[Function]}
    show={null}
    width="108px"
  >
    <pure(ArrowBackIosIcon) />
  </WithStyles(Component)>
  <WithStyles(Component)
    onClick={[Function]}
    show={null}
    width="108px"
  >
    <pure(ArrowForwardIosIcon) />
  </WithStyles(Component)>
  <WithStyles(Component)
    onClick={[Function]}
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "Add Category",
        }
      }
    />
  </WithStyles(Component)>
  <WithStyles(RawDecreaseMaxPoints)
    cancel={[Function]}
    deleteScorePoints={[Function]}
    open={false}
  />
  <WithStyles(RawDeleteScale)
    cancel={[Function]}
    deleteScale={[Function]}
    open={false}
    scaleIndex={0}
  />
  <WithStyles(RawDeleteTrait)
    cancel={[Function]}
    deleteTrait={[Function]}
    open={false}
  />
</div>
`;

exports[`Scale snapshot renders without standards 1`] = `
<div
  key="scale-0"
>
  <WithStyles(TraitsHeaderTile)
    currentPosition={0}
    key="header-key"
    maxPoints={2}
    onScaleChange={[Function]}
    onTraitLabelChange={[Function]}
    scaleIndex={0}
    scorePointsLabels={
      Array [
        "Non-Scorable",
        "Unsatisfactory",
        "Satisfactory",
      ]
    }
    scorePointsValues={Array []}
    setSecondaryBlockRef={[Function]}
    showDeleteScaleModal={[Function]}
    showStandards={false}
    traitLabel="Category"
    updateMaxPointsFieldValue={[Function]}
  />
  <DropTarget(DragSource(WithStyles(TraitTile)))
    currentPosition={0}
    index={0}
    key="0"
    onTraitChanged={[Function]}
    onTraitDropped={[Function]}
    onTraitRemoved={[Function]}
    scorePointsLabels={
      Array [
        "Non-Scorable",
        "Unsatisfactory",
        "Satisfactory",
      ]
    }
    scorePointsValues={Array []}
    secondaryBlockWidth={320}
    showStandards={false}
    trait={
      Object {
        "description": "",
        "name": "Presentation",
        "scorePointsDescriptors": Array [
          "Handwriting is unreadable, or response is blank, not in English, or too brief to evaluate. ",
          "Handwriting poor

Overall appearance is distracting to unacceptable",
          "Handwriting is generally legible

Overall appearance is acceptable or better",
        ],
        "standards": Array [],
      }
    }
  />
  <WithStyles(Component)
    left="232px"
    onClick={[Function]}
    show={null}
    width="108px"
  >
    <pure(ArrowBackIosIcon) />
  </WithStyles(Component)>
  <WithStyles(Component)
    onClick={[Function]}
    show={null}
    width="108px"
  >
    <pure(ArrowForwardIosIcon) />
  </WithStyles(Component)>
  <WithStyles(Component)
    onClick={[Function]}
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "Add Category",
        }
      }
    />
  </WithStyles(Component)>
  <WithStyles(RawDecreaseMaxPoints)
    cancel={[Function]}
    deleteScorePoints={[Function]}
    open={false}
  />
  <WithStyles(RawDeleteScale)
    cancel={[Function]}
    deleteScale={[Function]}
    open={false}
    scaleIndex={0}
  />
  <WithStyles(RawDeleteTrait)
    cancel={[Function]}
    deleteTrait={[Function]}
    open={false}
  />
</div>
`;

exports[`Scale snapshot renders without traits 1`] = `
<div
  key="scale-0"
>
  <WithStyles(TraitsHeaderTile)
    currentPosition={0}
    key="header-key"
    maxPoints={2}
    onScaleChange={[Function]}
    onTraitLabelChange={[Function]}
    scaleIndex={0}
    scorePointsLabels={
      Array [
        "Non-Scorable",
        "Unsatisfactory",
        "Satisfactory",
      ]
    }
    scorePointsValues={Array []}
    setSecondaryBlockRef={[Function]}
    showDeleteScaleModal={[Function]}
    showStandards={true}
    traitLabel="Category"
    updateMaxPointsFieldValue={[Function]}
  />
  <DropTarget(DragSource(WithStyles(TraitTile)))
    currentPosition={0}
    index={0}
    key="0"
    onTraitChanged={[Function]}
    onTraitDropped={[Function]}
    onTraitRemoved={[Function]}
    scorePointsLabels={
      Array [
        "Non-Scorable",
        "Unsatisfactory",
        "Satisfactory",
      ]
    }
    scorePointsValues={Array []}
    secondaryBlockWidth={320}
    showStandards={true}
    trait={
      Object {
        "description": "",
        "name": "Presentation",
        "scorePointsDescriptors": Array [
          "Handwriting is unreadable, or response is blank, not in English, or too brief to evaluate. ",
          "Handwriting poor

Overall appearance is distracting to unacceptable",
          "Handwriting is generally legible

Overall appearance is acceptable or better",
        ],
        "standards": Array [],
      }
    }
  />
  <WithStyles(Component)
    left="232px"
    onClick={[Function]}
    show={null}
    width="108px"
  >
    <pure(ArrowBackIosIcon) />
  </WithStyles(Component)>
  <WithStyles(Component)
    onClick={[Function]}
    show={null}
    width="108px"
  >
    <pure(ArrowForwardIosIcon) />
  </WithStyles(Component)>
  <WithStyles(Component)
    onClick={[Function]}
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "Add Category",
        }
      }
    />
  </WithStyles(Component)>
  <WithStyles(RawDecreaseMaxPoints)
    cancel={[Function]}
    deleteScorePoints={[Function]}
    open={false}
  />
  <WithStyles(RawDeleteScale)
    cancel={[Function]}
    deleteScale={[Function]}
    open={false}
    scaleIndex={0}
  />
  <WithStyles(RawDeleteTrait)
    cancel={[Function]}
    deleteTrait={[Function]}
    open={false}
  />
</div>
`;
